### PR TITLE
feat(aws): support configuring Lambda Handler

### PR DESCRIPTION
**Pull request by alexkli**
## Description

This adds a new `--aws-handler` argument that if set will be used for the `Handler` config of the AWS Lambda function.

```
--aws-handler  Set custom lambda Handler. For example, set if an AWS layer provides another function
               entry point.                                                                  [string]
```

## Motivation and Context

For instrumenting our Lambdas with New Relic using [their layer approach](https://docs.newrelic.com/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/instrument-your-own/) it is necessary to configure a different `Handler` on the Lambda function.

This is because the New Relic layer adds a "wrapper" handler for instrumentation purposes. This handler is called `newrelic-lambda-wrapper.handler`. Internally that will call the "actual" lambda code by looking at the configuration from the environment variable `NEW_RELIC_LAMBDA_HANDLER`, which needs to be set to `index.lambda` for a normal Helix deploy Lambda.

## How Has This Been Tested?

New unit test `aws.deployer.test.js > correctly uses awsHandler`:

* This validates the `Handler` value of the function config that is sent to the AWS Lambda [CreateFunction](https://docs.aws.amazon.com/lambda/latest/api/API_CreateFunction.html) API call.
* To do so I had to slightly refactor `AWSDeployer.js` so that it provides a `functionConfig` getter that can be inspected easily in unit tests without any network request mocking. Also used this to add some additional checks on existing tests.
* **Note:** I also tried a version where I did not refactor `AWSDeployer.js` but instead nock'ed all the HTTP requests done in `createLambda()`: https://github.com/adobe/helix-deploy/commit/7dcb08b7e78f52b096b565b38f911f014582af3f
  * However, this involves many requests and I think that this makes the tests harder to maintain in case the HTTP requests will change for some (unrelated) reason over time.
  * Feel free to pick whatever you prefer!

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by justinedelson** [comment]
@alexkli Can't this be accomplished by just setting the `AWS_LAMBDA_EXEC_WRAPPER` environment variable to `newrelic-lambda-wrapper.handler`?

**Comment by alexkli** [comment]
@justinedelson The [relevant New Relics docs](https://docs.newrelic.com/docs/serverless-function-monitoring/aws-lambda-monitoring/instrument-lambda-function/instrument-your-own/) didn't mention it. Let me test!

In any case I think it's still useful to be able to set it directly - also given the fact that hedy already dynamically sets it (if `esm` flag is on): https://github.com/adobe/helix-deploy/blob/2a3cdade2394854eb9cbb0fa20c579fbf997b348/src/deploy/AWSDeployer.js#L250

**Comment by alexkli** [comment]
Tested. It seems like `AWS_LAMBDA_EXEC_WRAPPER` does **not** work with the New Relic layer/extension. Errors with `newrelic-lambda-wrapper.handler: does not exist` (see below) and my actual Lambda code does not run at all.

But it works using hedy with this PR. Testing at https://github.com/adobe-rnd/assets-pool-newrelic-test/pull/14 (working revision: https://github.com/adobe-rnd/assets-pool-newrelic-test/pull/14/commits/bd1100cfe4ee04866b107e7c1a9a3dc0965de0f0)

New Relic layer logs:
```
17:19:28.480
INIT_START Runtime Version: nodejs:20.v27 Runtime Version ARN: arn:aws:lambda:us-east-1::runtime:672d5a3e06f81d120c089c5414b05186d7b4098504797c766bde2459847f38bc
 
17:19:28.582
[NR_EXT] New Relic Lambda Extension starting up
 
17:19:28.582
[NR_EXT] Initializing version 2.3.13 of the New Relic Lambda Extension...
 
17:19:28.595
newrelic-lambda-wrapper.handler: does not exist
 
17:19:28.596
EXTENSION Name: newrelic-lambda-extension State: Registered Events: [INVOKE, SHUTDOWN]
 
17:19:28.596
INIT_REPORT Init Duration: 116.02 ms Phase: init Status: error Error Type: Runtime.ExitError 
```

**Review by justinedelson — APPROVED** [review]

File: src/deploy/AWSDeployer.js [hunk 1]
```diff
@@ -218,9 +218,8 @@ export default class AWSDeployer extends BaseDeployer {
     this.log.info(chalk`{green ok:} deleted deploy package {blueBright s3://${this._bucket}/${this._key}}.`);
   }
 
-  async createLambda() {
+  get functionConfig() {
```
Comment (tripodsan) [inline src/deploy/AWSDeployer.js:221]: I don't like getters that contain too much logic, I would prefer an explicit:

```suggestion
  getFunctionConfig() {
```

**Review by tripodsan — APPROVED** [review]

**Pull request merged**